### PR TITLE
Fix Allow Scripts Once logic for embedded scripts and iframes.

### PR DESCRIPTION
**Pull request by goodov**
Fixed Allow Scripts Once to work for page-embedded scripts.

Resolves https://github.com/brave/brave-browser/issues/20744

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/content_settings/renderer/brave_content_settings_agent_impl.h [hunk 1]
```diff
@@ -99,9 +95,6 @@ class BraveContentSettingsAgentImpl
   // cache blocked script url which will later be used in `DidNotAllowScript()`
   GURL blocked_script_url_;
 
-  // temporary allowed script origins we preloaded for the next load
-  base::flat_set<std::string> preloaded_temporarily_allowed_scripts_;
```
Comment (goodov) [inline components/content_settings/renderer/brave_content_settings_agent_impl.h:103]: this doesn't work on reloads, an embedded script is executed before this is filled.

File: components/content_settings/renderer/brave_content_settings_agent_impl.cc [hunk 2]
```diff
@@ -342,7 +350,7 @@ bool BraveContentSettingsAgentImpl::AllowAutoplay(bool play_requested) {
 
 void BraveContentSettingsAgentImpl::SetAllowScriptsFromOriginsOnce(
     const std::vector<std::string>& origins) {
-  preloaded_temporarily_allowed_scripts_ = std::move(origins);
+  temporarily_allowed_scripts_ = origins;
```
Comment (goodov) [inline components/content_settings/renderer/brave_content_settings_agent_impl.cc:353]: `origins` is not movable (it's a mojo-generated method).

**Comment by goodov** [comment]
@darkdh @iefremov PTAL

**Review by darkdh — APPROVED** [review]
lgtm

**Comment by mkarolin** [comment]
Merging per @rebron

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build:

```
Brave | 1.37.10 Chromium: 98.0.4758.74 (Official Build) nightly (64-bit)
-- | --
Revision | d0fe1ec4df090cd3eb02b591228505e12ea476e9-refs/branch-heads/4758@{#935}
OS | Windows 11 Version 21H2 (Build 22000.438)
```

Used https://www.blizzard.com/en-gb for the following cases to ensure that both `Scripts blocked` & `Allow scripts once` are working as expected:

* ensured that scripts are correctly being blocked when `Scripts blocked` has been enabled
* ensured that the page loads correctly once  `Scripts blocked` has been disabled
* ensured that the correct state is maintained when restarting the browser or closing the tab/website
* ensured that clicking on down arrow near the number of scripts being blocked expands shields with more information
* ensured that `Scripts on this site` is correctly listing the scripts currently being blocked
* ensured that you can allow/block individual scripts via `Scripts on this site`
* ensured that `Allow all` works as expected under `Scripts on this site`
* ensured  `Allow scripts once` only works for that instance/tab (scripts should be blocked when loading the page again in a different tab if enabled)

**Pull request merged**